js/graviex.js

```diff
@@ -0,0 +1,991 @@
+        };
+        if (limit !== undefined) {
+            request['limit'] = limit; // default 100,
+        }
+        const response = await this.publicGetDepth (this.extend (request, params));
+        // {
+        //   "timestamp": 1643721617,
+        //   "asks": [
+        //     [
+        //       "0.00000253",
+        //       "120.0"
+        //     ],
+        //     [
+        //       "0.0000025",
+        //       "2424.0288"
+        //     ],
+        //     [
+        //       "0.00000249",
+        //       "77.577"
+        //     ],
+        //     [
+        //       "0.00000248",
+        //       "12.5"
+        //     ],
+        // },
+        const timestamp = this.safeInteger (response, 'timestamp');
+        const orderbook = this.parseOrderBook (response, symbol, timestamp);
+        orderbook['nonce'] = undefined;
+        return orderbook;
+    }
+
+    parseBalance (response) {
+        const balances = this.safeValue (response, 'accounts_filtered');
+        const result = { 'info': balances };
+        for (let i = 0; i < balances.length; i++) {
+            const balance = balances[i];
+            const currencyId = this.safeString (balance, 'currency');
+            const code = this.safeCurrencyCode (currencyId);
+            const account = this.account ();
+            account['free'] = this.safeString (balance, 'balance');
+            account['used'] = this.safeString (balance, 'locked');
+            result[code] = account;
+        }
+        return this.safeBalance (result);
+    }
+
+    async fetchBalance (params = {}) {
+        await this.loadMarkets ();
+        const response = await this.privateGetMembersMe (params);
+        // {
+        //     "sn": "GRAK6GDVXKNVIEX",
+        //     "name": null,
+        //     "email": "XXXXXXXXXXXXX",
+        //     "activated": true,
+        //     "verified": false,
+        //     "accounts_filtered": [
+        //     {
+        //         "currency": "gio",
+        //         "name": "GravioCoin",
+        //         "balance": "0.0",
+        //         "locked": "0.0",
+        //         "status": "ok",
+        //         "is_purged": false,
+        //         "is_coin": true
+        //     }
+        //     ]
+        // }
+        return this.parseBalance (response);
+    }
+
+    parseDepositAddress (depositAddress, currency = undefined) {
+        const code = (currency === undefined) ? undefined : currency['code'];
+        const tag = undefined;
+        this.checkAddress (depositAddress);
+        return {
+            'currency': code,
+            'address': depositAddress,
+            'tag': tag,
+            'network': undefined,
+            'info': depositAddress,
+        };
+    }
+
+    async fetchDepositAddress (code, params = {}) {
+        await this.loadMarkets ();
+        const currency = this.currency (code);
+        const request = {
+            'currency': currency['id'].toLowerCase (),
```

Comment:
why is it lowercase here? maybe the baseId and the quoteId should be lowercase instead

Reply:
@frosty00, currency parameter in the API endpoint supports the lower case values.

Reply:
Okay. Fixed in the latest commit https://github.com/ccxt/ccxt/pull/11697/commits/162b106bdb42084917d6187a3a60fb6e0f55e7d1

Reply:
it's not fixed, what I mean is to change the values of the `baseId` and the `quoteId` that are propagated in markets so they are lower case there

Reply:
@frosty00, Okay. Fixed in the latest commit https://github.com/ccxt/ccxt/pull/11697/commits/81c8152ac30b15da7aad6a3f117da386d51efd62